lib/client.js

```diff
@@ -15,29 +15,31 @@ function rawPostMessage(client, msg, forceReady) {
 }
 
 function wrappedPostMessage(name) {
-  var id = ++promiseCount;
-  var deferred = Promise.defer();
-  var params = Array.prototype.slice.call(arguments, 1);
+  var id = promiseCount++,
+      timeoutId,
+      params = Array.prototype.slice.call(arguments, 1);
 
-  // timeout the promise to ensure it will be garbage collected if nobody responds
-  var timeoutId = setTimeout(function() {
-    deferred.reject(new Error('Invocation request timeout'));
-  }, PROMISE_TIMEOUT);
+  var promise = new Promise(function(resolve, reject) {
+    // Time out the promise to ensure it will be garbage collected if nobody responds
+    timeoutId = setTimeout(function() {
+      reject(new Error('Invocation request timeout'));
+    }, PROMISE_TIMEOUT);
 
-  // ensure promise is cleaned up when resolved
-  deferred.promise.then(removePromise.bind(null, id, timeoutId), removePromise.bind(null, id, timeoutId));
-
-  pendingPromises[id] = deferred;
+    pendingPromises[id] = {resolve: resolve, reject: reject};
 
-  var msg = JSON.stringify({ id: id, request: name, params: params, appGuid: this._appGuid });
-  rawPostMessage(this, msg);
+    var msg = JSON.stringify({ id: id, request: name, params: params, appGuid: this._appGuid });
+    rawPostMessage(this, msg);
+  }.bind(this));
 
-  return deferred.promise;
+  // ensure promise is cleaned up when resolved
+  return promise.then(removePromise.bind(null, id, timeoutId), removePromise.bind(null, id, timeoutId));
 }
 
-function removePromise(id, timeoutId) {
+function removePromise(id, timeoutId, args) {
   clearTimeout(timeoutId);
   delete pendingPromises[id];
+  if (args instanceof Error) throw args;
```

Comment:
This is so the error bubbles to any chained promise handlers in the event that the promise being removed fails?